### PR TITLE
fix: never assume an unidentified Latin language is English; clamp temperatures (#35)

### PR DESCRIPTION
**Pull request by avinashk369**
Fixes two independent defects reported in #35, where Laya was evaluated as a 13-way skill router on Romanian agent traffic. Both are code-only and need no retraining.

Thanks to @Skystar9999 for the report and the per-case JSON — every claim in it reproduced, and the JSON is what made the second defect diagnosable.

## 1. `Router` silently sent Romanian to the English checkpoint

`analyse()` computed `is_english` as `lang in (None, "en")`, so **undecided counted as English**. `_STOP` covers 7 languages, which means every other Latin-script language — Romanian, Polish, Czech, Turkish, Vietnamese, Indonesian — was handed to the checkpoint that scores 0.330 accuracy with 0.658 ECE on `ro`, instead of the multilingual one. Three of the four Romanian states in #35 routed to English:

```
'Care este ora in Tokyo?'        -> language='en',  is_english=True   -> ENGLISH
'Cât e ora acum în Tokyo?'       -> language='fr',  is_english=False  -> multilingual (by luck)
'Gătește-mi o rețetă de sarmale' -> language=None,  is_english=True   -> ENGLISH
'Exportă APK-ul pentru Android'  -> language=None,  is_english=True   -> ENGLISH
```

Anyone following the README quickstart with `Router(preload=True)` on Romanian got the worse model with no signal that it happened. #23 was called the same class of bug but only extended the *script* table, so this Latin-script fallback was still open.

**What changed.** Undecided is now its own answer. When nothing identifies the language, non-English letters are enough to prefer the multilingual checkpoint; text without them (including short English) still goes to English:

```python
undecided = lang is None
english = lang == "en" or (undecided and not prof_lat["looks_non_english"])
```

Supporting changes:

- `latin_profile()` exposes the evidence (`english_hits`, `diacritic_rate`, `looks_non_english`), because `analyse` needs the evidence and not just the verdict.
- `_NON_EN_DIACRITICS` now covers Romanian, Polish, Czech/Slovak, Hungarian, Turkish and Baltic letters. This is the signal that catches a language with no stopword list at all, so it generalises past Romanian.
- A 0–0 tie between non-English stopword lists no longer names a winner. That invention is what reported Romanian as French, and a single shared function word (`para` in Turkish text) is no longer enough to claim Spanish.
- Romanian stopwords added, deliberately omitting `la` / `o` / `un` / `de` / `pe` / `ca` so `ro` cannot steal a French, Spanish, Italian or Portuguese state.
- `Router.route()` now explains the undecided case honestly instead of printing `looks like None`:
  `Latin script, language not identified but 10% non-English letters; not safe for the English checkpoint`
- Every branch of `analyse()` returns the same keys, so callers can read `diacritic_rate` without guarding.

**One case is still wrong, on purpose, and is pinned by a test named `latin/KNOWN GAP romanian without diacritics`:** `"Care este ora in Tokyo?"` is short enough to carry no diacritics and `in` is a real English function word. Relaxing the margin rule to catch it would put ordinary English at risk, which is the property the current heuristic is built to protect. A real LID model (fastText-class, ~1 ms as @Skystar9999 notes) is still the right fix — this removes the *silent* failure in the meantime and leaves that work clearly scoped.

## 2. A fitted temperature of 0.1006 published 0.99 confidence on a 0.24 belief

This is the part of #35 that I think was under-read, including in my own first pass. The two checkpoints fail for **opposite** reasons:

```
english          temperature_by_options={'choice:3-5': 1.760, 'choice:11+': 0.1006, ...}
multilingual     temperature_by_options={}                     temperature=[1.0, 1.0, 1.0]
typed-decisions  temperature_by_options={'choice:3-5': 1.760, 'choice:11+': 0.1006, ...}
```

`multilingual` is uncalibrated — raw softmax, as BENCHMARKS.md says. But `english` and `typed-decisions` are **anti-calibrated on precisely the bucket routing uses**: 13 options lands in `choice:11+`, and dividing logits by 0.1006 sharpens them ~10x.

```
temp_bucket(choice, 13) -> 'choice:11+' -> T = 0.1006
T=1.0000  top-prob=0.2378
T=0.1006  top-prob=0.9931
```

That is why the reporter's `typed_decisions_en` run shows mean confidence 0.96 on correct answers **and 0.81 on wrong ones**, with four wrong answers above 0.85. Computed from their per-case JSON:

| run | accuracy | conf when right | conf when wrong | separation |
|---|---|---|---|---|
| `multilingual_en` | 50% | 0.82 | 0.80 | +0.01 |
| `multilingual_ro` | 50% | 0.66 | 0.61 | +0.06 |
| `typed_decisions_en` | 55% | 0.96 | 0.81 | +0.15 |

**What changed.** `clamp_temperature()` in `laya/common.py`, applied at load in `Agent.__init__`, confining temperatures to `[0.5, 5.0]` and falling back to 1.0 for `None`/NaN/inf/garbage. The previous guard was `max(1e-3, float(t_scale))`, which never fired. The raw values stay on the Agent as `temperature_raw` / `temperature_by_options_raw` so a calibration harness can still read what shipped, and a `RuntimeWarning` names exactly what was rejected:

```
RuntimeWarning: laya: this checkpoint ships temperatures outside [0.5, 5] which would
distort confidence; clamping choice:11+=0.1006. Treat confidence from the affected
buckets as uncalibrated.
```

Measured on the real `typed-decisions` weights at 13 options:

| case | pick | T=0.1006 | T=0.5 |
|---|---|---|---|
| what time is it in Tokyo | niciunul | 1.0000 | 0.9187 |
| loop-play mp4s | niciunul | 0.9750 | **0.5542** |
| translate to french | niciunul | 1.0000 | 0.8124 |

Row two is a wrong pick in the reporter's gold labels. It would have passed a `min_confidence=0.85` gate at 0.975 and now fails it at 0.554.

**Clamping cannot change which option is chosen** — dividing logits by a positive scalar preserves the argmax — so no accuracy number in BENCHMARKS.md can move. Only calibration does.

## Verification

| check | result |
|---|---|
| `tests/test_router.py` | **144 passed, 0 failed** (was 106) |
| `tests/test_criteria.py` | 34 passed, 0 failed |
| `ruff` + `compileall` | clean |
| `python -m build` + `twine check` | both artifacts pass |
| `tests/test_local_e2e.py`, real weights, all three checkpoints | **23 passed, 0 failed** |

The 38 new tests cover Romanian, Polish, Czech, Turkish and Vietnamese routing; the English non-regressions (short states, loanwords like `cafe`/`naive`); undecided being reported as undecided rather than dressed up as a detection; the clamp's boundaries and its NaN/inf/garbage fallbacks; and a stable key set from `analyse()` on every branch.

## Deliberately not in scope

- Refitting or dropping the `choice:11+` bucket in the checkpoints themselves — needs refitting, not a code change. The clamp is a guardrail, not calibration.
- Shipping fitted temperatures for `multilingual` — waiting on #19 (`fit_temperatures` / `save_calibration` / `load_calibration`). Worth noting #19 is tooling and does not by itself fix either defect here.
- The README warning that `laya-multilingual` is uncalibrated as shipped, and accepting the community eval harness offered in #35.

### CODE REVIEW

**Comment by NandhaKishorM** [comment]
Thank you for this thorough PR. Clamping temperature_by_options to [0.5, 5.0] prevents extreme logit sharpening, and using diacritic rate to detect unlisted Latin languages avoids misrouting them to the English checkpoint.
